### DISCUSSION TITLE
Question: How to get Context?

### DISCUSSION THREAD

**Discussion by brunocalou** [pmndrs/jotai#1029] [Q&A]
Hello guys, I love Jotai and I've been using it on a huge project of mine for a while.

> TLDR - I've come to a use case where I need to access the Context directly and I found no way of doing it.

Here's the full context:

I'm developing a component using the `@react-three/fiber` and `@react-three/drei` packages. When I use the `<Canvas />` component, all the `Context` in React is lost. To solve it, I have to use the `<ContextBridge />` component from `@react-three/drei` package. It looks like this:

```typescript
export const MyComponent = (props: MyComponentPropsProps) => {
  const ContextBridge = useContextBridge(ThemeContext); // Get ThemeContext and pass to elements inside <Canvas />
  const myState = useAtomValue(myStateAtom); // Get state from Jotai context

  return (
    <Canvas gl={{ antialias: false }} linear flat>
      <ContextBridge>
        <MyInnerComponent
          myState={myState}
        />
      </ContextBridge>
    </Canvas>
  );
};
```

I want to access the state on Jotai Provider without passing it as a prop to my inner components. In order to do it, theoretically, I would have to get the context and pass it as a parameter on `useContextBridge`, like so

 ```typescript
 const ContextBridge = useContextBridge(ThemeContext, JotaiContext);
```

I can use Jotai hooks on inner components inside the Canvas, but the context is not the same, it's a brand new one.

How can I keep the same context inside the Canvas component? Is there a better way of doing it?

Here is a [Codesandbox link](https://codesandbox.io/s/silly-aryabhata-5gr6sr?file=/src/App.tsx) reproducing the issue

Thanks in advance!

**Comment by dai-shi** [comment]
Context is hidden as an implementation detail and not exposed as a public api.
That said, I know the ContextBridge thing and why it's necessary with r3f. (we used to have a context bridge before v1.)
The solution we came up with is `unstable_createStore`.
Please refer #861 and #922, and if you are not sure how to use it, feel free to ask here.
